### PR TITLE
Implement BF formatting using the new options module

### PR DESCRIPTION
**Pull request by Spaak**
As discussed in #101.

The lower-level functions `bayesian.bayesfactor_ttest` etc. now return `float`s; all post-processing (i.e., formatting into scientific or positional string) is handled at the DataFrame level in `_postprocess_dataframe`.

### CODE REVIEW

File: pingouin/bayesian.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 def _format_bf(bf, precision=3, trim='0'):
     """Format BF10 to floating point or scientific notation.
     """
+    if type(bf) == str:
```
Comment (raphaelvallat) [inline pingouin/bayesian.py:13]: Why is this needed?

**Comment by raphaelvallat** [comment]
Great! I just added one comment on the code. Can you also please update the changelog? Thanks!

File: pingouin/bayesian.py (hunk 1 above)
Comment (Spaak) [reply pingouin/bayesian.py:13]: `pairwise.pairwise_ttests` internally calls `parametric.ttest`, which also returns a post-processed DataFrame. At the end of `pairwise_ttests`, `postprocess_dataframe` is called again, but we don't want to attempt to format the BF again (which is by then already a string).

**Comment by Spaak** [comment]
> Can you also please update the changelog? Thanks!

Done!

**Review by raphaelvallat — APPROVED** [review]

**Comment by raphaelvallat** [comment]
Justed merged the PR. Thank you!
